plugins/inputs/intel_rdt/intel_rdt.go

```diff
@@ -0,0 +1,552 @@
+	mapProcessPIDs := map[string]string{}
+
+	availableProcesses, err := r.Processor.getAllProcesses()
+	if err != nil {
+		return nil, fmt.Errorf("cannot gather information of all available processes")
+	}
+	for _, availableProcess := range availableProcesses {
+		if choice.Contains(availableProcess.Name, providedProcesses) {
+			PID := availableProcess.PID
+			mapProcessPIDs[availableProcess.Name] = mapProcessPIDs[availableProcess.Name] + fmt.Sprintf("%d", PID) + ","
+		}
+	}
+	for key := range mapProcessPIDs {
+		mapProcessPIDs[key] = strings.TrimSuffix(mapProcessPIDs[key], ",")
+	}
+	return mapProcessPIDs, nil
+}
+
+func (r *IntelRDT) createArgsAndStartPQOS(ctx context.Context) {
+	args := []string{"-r", "--iface-os", "--mon-file-type=csv", fmt.Sprintf("--mon-interval=%d", r.SamplingInterval)}
+
+	if len(r.parsedCores) != 0 {
+		coresArg := createArgCores(r.parsedCores)
+		args = append(args, coresArg)
+		go r.readData(args, nil, ctx)
+
+	} else if len(r.processesPIDsMap) != 0 {
+		processArg := createArgProcess(r.processesPIDsMap)
+		args = append(args, processArg)
+		go r.readData(args, r.processesPIDsMap, ctx)
+	}
+	return
+}
+
+func (r *IntelRDT) readData(args []string, processesPIDsAssociation map[string]string, ctx context.Context) {
+	r.wg.Add(1)
+	defer r.wg.Done()
+
+	cmd := exec.Command(r.PqosPath, append(args)...)
+
+	cmdReader, err := cmd.StdoutPipe()
+	if err != nil {
+		r.errorChan <- err
+	}
+	go r.processOutput(cmdReader, processesPIDsAssociation)
+
+	go func() {
+		for {
+			select {
+			case <-r.stopPQOSChan:
+				if err := shutDownPqos(cmd); err != nil {
+					r.Log.Error(err)
+				}
+				return
+			case <-ctx.Done():
+				if err := shutDownPqos(cmd); err != nil {
+					r.Log.Error(err)
+				}
+				return
+			}
+		}
+	}()
+	err = cmd.Start()
+	if err != nil {
+		r.errorChan <- fmt.Errorf("pqos: %v", err)
+		return
+	}
+	err = cmd.Wait()
+	if err != nil {
+		r.errorChan <- fmt.Errorf("pqos: %v", err)
+	}
+}
+
+func (r *IntelRDT) processOutput(cmdReader io.ReadCloser, processesPIDsAssociation map[string]string) {
+	reader := bufio.NewScanner(cmdReader)
+	/*
+		Omit constant, first 4 lines :
+		"NOTE:  Mixed use of MSR and kernel interfaces to manage
+				CAT or CMT & MBM may lead to unexpected behavior.\n"
+		CMT/MBM reset successful
+		"Time,Core,IPC,LLC Misses,LLC[KB],MBL[MB/s],MBR[MB/s],MBT[MB/s]\n"
+	*/
+	toOmit := pqosInitOutputLinesNumber
+
+	// omit first measurements which are zeroes
+	if len(r.parsedCores) != 0 {
+		toOmit = toOmit + len(r.parsedCores)
+		// specify how many lines should pass before stopping
+	} else if len(processesPIDsAssociation) != 0 {
+		toOmit = toOmit + len(processesPIDsAssociation)
+	}
+	for omitCounter := 0; omitCounter < toOmit; omitCounter++ {
+		reader.Scan()
+	}
+	for reader.Scan() {
+		out := reader.Text()
+		// to handle situation when monitored PID disappear and "err" is shown in output
+		if strings.Contains(out, "err") {
```

Comment:
err seems like a common string segment. any chance it catches non-error text?

Reply:
Good point. But pqos tool basically returns only numerical-like data in form of CSV. 
There is an example of pqos output:

**Format:**
`Time,PID,Core,IPC,LLC Misses,LLC[KB],MBL[MB/s],MBR[MB/s],MBT[MB/s]`

**Without errorr:**
`2020-08-12 13:34:36,"45417,29170",37,44,0.00,0,0.0,0.0,0.0,0.0`

**With error:**
`2020-08-12 13:34:36,"45417,29170",err,0.00,0,0.0,0.0,0.0,0.0`

The `err` substring in `Core` field is a very specific case showing only when some problem with the monitored process happened.